tests/test_paperscraper.py

```diff
@@ -7,27 +7,116 @@
+
+
+def test_format_bibtex_badkey():
+    bibtex1 = """
+            @article{Moreira2022Safety,
+            title        = {Safety and Efficacy of a Third Dose of BNT162b2 Covid-19 Vaccine},
+            volume       = {386},
+            ISSN         = {1533-4406},
+            url          = {http://dx.doi.org/10.1056/nejmoa2200674},
+            DOI          = {10.1056/nejmoa2200674},
+            number       = {20},
+            journal      = {New England Journal of Medicine},
+            publisher    = {Massachusetts Medical Society},
+            author       = {Moreira, Edson D. and Kitchin, Nicholas and Xu, Xia and Dychter, Samuel S. and Lockhart, Stephen and Gurtman, Alejandra and Perez, John L. and Zerbini, Cristiano and Dever, Michael E. and Jennings, Timothy W. and Brandon, Donald M. and Cannon, Kevin D. and Koren, Michael J. and Denham, Douglas S. and Berhe, Mezgebe and Fitz-Patrick, David and Hammitt, Laura L. and Klein, Nicola P. and Nell, Haylene and Keep, Georgina and Wang, Xingbin and Koury, Kenneth and Swanson, Kena A. and Cooper, David and Lu, Claire and Türeci, Özlem and Lagkadinou, Eleni and Tresnan, Dina B. and Dormitzer, Philip R. and Şahin, Uğur and Gruber, William C. and Jansen, Kathrin U.},
+            year         = {2022},
+            month        = may,
+            pages        = {1910-1921}
+            }
+            """  # noqa: E501
+    assert format_bibtex(bibtex1, "Moreira2022Safety", clean=False)
 
 
 class Test0(IsolatedAsyncioTestCase):
     async def test_google_search_papers(self):
         query = "molecular dynamics"
         papers = await paperscraper.a_search_papers(
-            query, search_type="google", year="2019-2023"
+            query, search_type="google", year="2019-2023", limit=5
         )
-        assert len(papers) >= 1
+        assert len(papers) >= 3
 
         query = "molecular dynamics"
         papers = await paperscraper.a_search_papers(
-            query, search_type="google", year="2020"
+            query, search_type="google", year="2020", limit=5
         )
-        assert len(papers) >= 1
+        assert len(papers) >= 3
 
         query = "covid vaccination"
         papers = await paperscraper.a_search_papers(query, search_type="google")
-        assert len(papers) >= 1
+        assert len(papers) >= 3
+
+
+class TestGS(IsolatedAsyncioTestCase):
+    async def test_gsearch(self):
+        query = "molecular dynamics"
+        papers = await paperscraper.a_gsearch_papers(query, year="2019-2023", limit=3)
+        print(papers)
+        assert len(papers) >= 3
+
+        query = "molecular dynamics"
+        papers = await paperscraper.a_gsearch_papers(query, year="2020", limit=3)
+        assert len(papers) >= 3
+
+        query = "covid vaccination"
+        papers = await paperscraper.a_gsearch_papers(query, limit=3)
+        assert len(papers) >= 3
+
+        # check their details
+        for paper in papers.values():
+            print(paper)
```

Comment:
Feel free to ignore this, but I find `print`s in test lead to log spam

Reply:
this is left over - thanks